### DISCUSSION TITLE
social account: youtube link recognition

### DISCUSSION THREAD

**Discussion by AlexanderLanin** [community/community#53909] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I'm attempting to add a social account youtube link, but it's only being recognized when it contains a "user/" within the url. However then it also shows "user/" within the link text, which isn't super pretty.

Both links lead to the exact same page, but one is not recognized as youtube, and the other has a "user/" prefix.
![image](https://user-images.githubusercontent.com/5074553/234969294-708dae04-a1f9-4e35-b53e-5a79ebeedd89.png)

It's even visible via `/users/USER/social_accounts` endpoint, that the first one is `generic`:

```
[
  {
    "provider": "generic",
    "url": "https://www.youtube.com/x/"
  },
  {
    "provider": "youtube",
    "url": "https://www.youtube.com/user/x"
  }
]
```

**Comment by davidjbrossard** [comment]
I am facing the same issue. I want to add a channel that is not a user but it's not recognized as YouTube.

**Comment by Dineth-H** [comment]
I understand your frustration. There are a few ways to fix the issue of your YouTube link showing the "user/" prefix:

- Use a **link shortener**. A link shortener is a service that allows you to shorten long URLs. This can be a good way to get rid of the "user/" prefix in your YouTube link. Some popular link shorteners include Bitly, TinyURL, and Rebrandly.

- Use a **custom URL**. YouTube allows you to create a custom URL for your channel. This is a shorter, more memorable URL that doesn't contain the "user/" prefix. To create a custom URL, go to your YouTube channel settings and click on the "Advanced" tab.

- Contact GitHub support. If you're still having trouble, you can contact GitHub support for help. They may be able to help you troubleshoot the issue or provide you with a workaround.

**Here are the steps to create a short link using Bitly:**

1. Go to the Bitly website and create an account.
2. Once you're logged in, click the "Create" button.
3. Paste your YouTube link into the "Long URL" field and click the "Shorten" button.
4. Bitly will generate a short link for you.
5. Copy the short link and paste it into your GitHub profile.

**Here are the steps to create a custom URL for your YouTube channel:**

1. Go to your YouTube channel settings and click on the "Advanced" tab.
2. Under the "Custom URL" section, click the "Change" button.
3. Enter the custom URL that you want to use and click the "Update" button.

I hope this helps!

**Reply by Avasam** [reply]
A URL shortener or custom URL wouldn't recognize the link as YouTube and add the icon.
Workarounds are already listed, having to accept either:
- A long url, no icon
- A Youtube icon, weird looking `user/` prefix but working link

It's a simple UI bug likely caused by YouTube's recent channel URL changes. No help needed, just a visual bug to report (if there's a better place to report the bug though, I'd be happy to, or upvote the proper bug report)

**Comment by ghost** [comment]
Response #1
